src/dex/uniswap-v2/ring-v2.ts

```diff
@@ -0,0 +1,371 @@
+        ? await Promise.all(
+            amountsWithFee.map(amount =>
+              this.getSellPricePath(amount, [pairParam]),
+            ),
+          )
+        : await Promise.all(
+            amountsWithFee.map(amount =>
+              this.getBuyPricePath(amount, [pairParam]),
+            ),
+          );
+
+      const [unitOutWithFee, ...outputsWithFee] = applyTransferFee(
+        [unit, ...prices],
+        side,
+        // This part is confusing, because we treat differently SELL and BUY fees
+        // If Buy, we should apply transfer fee on srcToken on top of dexFee applied earlier
+        // But for Sell we should apply only one dexFee
+        isSell ? transferFees.destDexFee : transferFees.srcFee,
+        isSell ? this.DEST_TOKEN_DEX_TRANSFERS : SRC_TOKEN_PARASWAP_TRANSFERS,
+      );
+
+      // As ringv2 just has one pool per token pair
+      return [
+        {
+          prices: outputsWithFee,
+          unit: unitOutWithFee,
+          data: {
+            router: this.router,
+            path: [from.address.toLowerCase(), to.address.toLowerCase()],
+            factory: this.factoryAddress,
+            initCode: this.initCode,
+            feeFactor: this.feeFactor,
+            pools: [
+              {
+                address: pairParam.exchange,
+                fee: parseInt(pairParam.fee),
+                direction: pairParam.direction,
+              },
+            ],
+          },
+          exchange: this.dexKey,
+          poolIdentifier,
+          gasCost: this.poolGasCost,
+          poolAddresses: [pairParam.exchange],
+        },
+      ];
+    } catch (e) {
+      this.logger.debug('Error in getPricesVolume');
+
+      if (blockNumber === 0)
+        this.logger.error(
+          `Error_getPricesVolume: Aurelius block manager not yet instantiated`,
+        );
+      this.logger.error(`Error_getPrices:`, e);
+      return null;
+    }
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: UniswapData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    let args: any[];
+    let functionName: RingV2Functions;
+    let specialDexFlag: SpecialDex;
+    let transferSrcTokenBeforeSwap: Address | undefined;
+    let targetExchange: Address;
+    let dexFuncHasRecipient: boolean;
+
+    let ttl = 1200;
+    const deadline = `0x${(
+      Math.floor(new Date().getTime() / 1000) + ttl
+    ).toString(16)}`;
+
+    const [from, to] = this.getTokenAddresses(srcToken, destToken);
+    let path: Address[] = [from.toLowerCase(), to.toLowerCase()];
+
+    if (isETHAddress(srcToken)) {
+      if (side == SwapSide.SELL) {
+        functionName = RingV2Functions.swapExactETHForTokens;
+        args = [destAmount, path, recipient, deadline];
+      } else {
+        functionName = RingV2Functions.swapETHForExactTokens;
+        args = [srcAmount, path, recipient, deadline];
+      }
```

Comment:
not clear to me why we are passing `destAmount` on SELL instead of `srcAmount`. At the same time, we always handle wrapped tokens on uni-v2, so no need in covering ETH case. Therefore, I suggest to remove ETH related part and leave only `swapExactTokensForTokens` and `swapTokensForExactTokens`

Reply:
also, ETH related methods have [almost no usage](https://bloxy.info/address/0x39d1d8fcC5E6EEAf567Bce4e29B94fec956D3519), so not relevant at least in terms of this